### PR TITLE
Support import disk functionallity

### PR DESCRIPTION
**Pull request by jacobcrawford**
Added support for the wic format. The wic format is imported as a disk that is then attached to the VM. This is done through remote ssh provisioning.

### CODE REVIEW

File: vm_module/main.tf [hunk 1]
```diff
@@ -11,15 +11,17 @@ resource "proxmox_vm_qemu" "node"  {
     name = var.name
     desc = "A mukube test node"
     target_node = var.target_node
-    iso = var.iso
+    iso = var.os_format == "iso" ? var.os : "/var/lib/vz/template/iso/ubuntu-18.04.5-desktop-amd64.iso" ## We need to specify iso
```
Comment (skovbakke) [inline vm_module/main.tf:14]: Why do we default to ubuntu desktop?

**Review by skovbakke — CHANGES_REQUESTED** [review]
We should not use the absolute path of files.

File: vm_module/main.tf (hunk 1 above)
Comment (jacobcrawford) [reply vm_module/main.tf:14]: To use the provider one of "clone" and "iso" needs to be specified. I just put Ubuntu to get it to work, but we need a different solution. Though it has to be a file that we know will not be deleted.

File: variables.tf [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
 }
 
 variable "allow_tls_insecure" {
   type = bool
 }
 
-variable "iso_path" {
+variable "os_path" {
+  type = string
+  description = "The absolute path to the os folder in the Proxmox."
```
[the remark this answers is not in the document]
Comment (jacobcrawford) [reply variables.tf:18]: Unfortunately, this cannot be done for wic image files.

File: vm_module/main.tf (hunk 1 above)
Comment (jacobcrawford) [reply vm_module/main.tf:14]: I have downloaded an Alpine image instead.

File: vm_module/main.tf [hunk 3]
```diff
@@ -36,3 +38,30 @@ resource "proxmox_vm_qemu" "node"  {
...
+  }
+
+  provisioner "file" {
+    source = "script.sh" 
+    destination = "/tmp/script.sh"
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+      "chmod +x /tmp/script.sh",
```
Comment (skovbakke) [inline vm_module/main.tf:59]: How about making the script executable?  Currently the script access mode is `664`. or does the mode somehow change after it is transferred to the remote machine?

**Review by skovbakke — CHANGES_REQUESTED** [review]
looks good, only a small suggestion left.

File: vm_module/main.tf (hunk 1 above)
Comment (skovbakke) [reply vm_module/main.tf]: That is probably better.

File: script.sh [hunk 4]
```diff
@@ -0,0 +1,11 @@
+VM_ID=$1
```
Comment (klausenbusk) [inline script.sh:1]: Please add a shebang:
```suggestion
#!/bin/bash
VM_ID=$1
```

File: script.sh [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+IMAGE_PATH=$2
+STORAGE_POOL=$3
+# The new imported disk will get a name that depends on the current number of disks for the vm 
+DISK_COUNT=$4
+
+qm stop $VM_ID
+qm importdisk $VM_ID $IMAGE_PATH $STORAGE_POOL
+qm set $VM_ID --scsi0 $STORAGE_POOL:vm-$VM_ID-disk-$DISK_COUNT
+qm set $VM_ID --boot order=scsi0
+qm start $VM_ID
```
Comment (klausenbusk) [inline script.sh:11]: Fix [SC2086](https://github.com/koalaman/shellcheck/wiki/SC2086):
```suggestion
qm stop "$VM_ID"
qm importdisk "$VM_ID" "$IMAGE_PATH $STORAGE_POOL"
qm set "$VM_ID" --scsi0 "$STORAGE_POOL:vm-$VM_ID-disk-$DISK_COUNT"
qm set "$VM_ID" --boot order=scsi0
qm start "$VM_ID"
```

File: vm_module/main.tf (hunk 3 above)
Comment (jacobcrawford) [reply vm_module/main.tf:59]: As the script is only supposed to be executable on the remote machine I think it is best practice to leave it non-executable in the version control. Maybe we should restrict it further in the version control.

Comment (skovbakke) [reply vm_module/main.tf:59]: That is fine

**Pull request merged**